bin/manage.py

```diff
@@ -0,0 +1,406 @@
+import json
+import logging
+import os
+import socket
+import signal
+import struct
+import sys
+import time
+
+from functools import wraps
+
+import consul as pyconsul
+#import manta
+from pymongo import MongoClient
+from pymongo.errors import *
+
+consul = pyconsul.Consul(host=os.environ.get('CONSUL', 'consul'))
+logging.basicConfig(format='%(asctime)s %(levelname)s %(name)s %(message)s',
+                    stream=sys.stdout,
+                    level=logging.getLevelName(
+                        os.environ.get('LOG_LEVEL', 'INFO')))
+
+log = logging.getLogger('manage.py')
+
+def debug(fn):
+    """
+    Function/method decorator to trace calls via debug logging.
+    Is a pass-thru if we're not at LOG_LEVEL=DEBUG. Normally this
+    would have a lot of perf impact but this application doesn't
+    have significant throughput.
+    """
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        try:
+            # because we have concurrent processes running we want
+            # to tag each stack with an identifier for that process
+            arg = "[{}]".format(sys.argv[1])
+        except IndexError:
+            arg = "[pre_start]"
+        name = '{}{}{}'.format(arg, (len(inspect.stack()) * " "), fn.__name__)
+        log.debug('%s' % name)
+        out = apply(fn, args, kwargs)
+        log.debug('%s: %s', name, out)
+        return out
+    return wrapper
+
+def get_environ(key, default):
+    """
+    Gets an environment variable and trims away comments and whitespace.
+    """
+    val = os.environ.get(key, default)
+    try:
+        val = val.split('#')[0]
+        val = val.strip()
+    finally:
+        # just swallow AttributeErrors for non-strings
+        return val
+
+# ---------------------------------------------------------
+
+SESSION_CACHE_FILE = get_environ('SESSION_CACHE_FILE', '/tmp/mongodb-session')
+SESSION_NAME = get_environ('SESSION_NAME', 'mongodb-replica-set-lock')
+SESSION_TTL = int(get_environ('SESSION_TTL', 60))
+
+# consts for node state
+PRIMARY = 'mongodb-primary'
+SECONDARY = 'mongodb-secondary'
+REPLICA = 'mongodb'
+
+# key where primary will be stored in consul
+PRIMARY_KEY = get_environ('PRIMARY_KEY', 'mongodb-primary')
+
+# how many time to retry connecting to mongo
+# be aware that the health check may timeout before this is reached
+MONGO_RETRY_TIMES=int(get_environ('MONGO_RETRY_TIMES', 10))
+
+# ---------------------------------------------------------
+
+class ContainerPilot(object):
+    """
+    ContainerPilot config is where we rewrite ContainerPilot's own config
+    so that we can dynamically alter what service we advertise
+    """
+
+    def __init__(self):
+        # TODO: we should make sure we can support JSON-in-env-var
+        # the same as ContainerPilot itself
+        self.path = get_environ('CONTAINERPILOT', None).replace('file://', '')
+        with open(self.path, 'r') as f:
+            self.config = json.loads(f.read())
+
+    @debug
+    def update(self, state):
```

Comment:
That makes sense, I was mostly just following the mysql setup.  I am currently testing it with no other containerpilot "states" (and thus no reloads :tada:), but I fear we may need something tracking when a node is uninitialized, `startup2`, or `recovering`. I think mongo clients will be able to get to the primary node from a `startup2` or `recovering` node.